automation/test.sh

```diff
@@ -2,7 +2,17 @@
 
 main() {
     ls -l
+    install_promtool
+
     echo "Running tests"
+    ./promtool test rules test/promql/*
+}
+
+install_promtool() {
+    echo "Installing promtool..."
+    VERSION="2.42.0"
+    curl -OJL "https://github.com/prometheus/prometheus/releases/download/v${VERSION}/prometheus-$VERSION.linux-amd64.tar.gz"
```

Comment:
Do we really have to curl something for each and every CI run? Can we arrange for this to be preinstalled in whatever image we are using to run this?

Frankly I'm not sure why is there a shell script here at all, shouldn't we be doing things via Tekton tasks?

Reply:
this script is being called as part of a tekton task. 
going forward we will probably have an image with all dependencies, but for now, I think this PR is good enough.

Reply:
why run a script rather then write the thing we need to run directly in the task code? what do we gain from adding a level of indirection here?

Reply:
ide support, shellcheck, having the ability to run it locally without tekton.